### ISSUE TITLE
Add differentiation between undirected and directed Graphs

### ISSUE THREAD

**Issue by rolyhudson** [BHoM/BHoM_Engine#1662]
Right now `Graph` in the BHoM is directed. We should provide the functionality to allow these to be undirected. Possibly just by adding a property like `IsDirected` to the `Graph` or perhaps on the `GraphLink`.

This came up in the PR discussion https://github.com/BHoM/BHoM_Engine/pull/1637#discussion_r405470610

**Comment by epignatelli** [comment]
I think that, if we want to take this road, the `IsDirected` should be a `Query` method, because it is the result of the graph connectivity - `Query.IsDirected(this Graph)`.
This put us in the position, though, that every time we write a method, we assume to work on an undirected graph.

**Comment by rolyhudson** [comment]
My understanding is that an undirected graph consists of links with unordered pairs of nodes and a directed graph consists of links with ordered pairs of nodes. 

At present the BHoM `GraphLink` has a `StartNode` and `EndNode`, to me those names imply it is an ordered pair of nodes and the graph is directed by default.  

I was thinking we could adapt the current directed graph to be undirected by forcing the ordering of link nodes to be disregarded using a property on the Graph. Or with another `GraphLink` object where the nodes are named to indicate they are unordered.

**Comment by epignatelli** [comment]
I agree on all the above @rolyhudson. Adding a new property to `GraphLink` seems like the way to go to me. The `IsDirected` method, could just check if all the links are undirected.
